src/huggingface_hub/hf_api.py

```diff
@@ -1615,46 +1615,108 @@ def run_as_future(self, fn: Callable[..., R], *args, **kwargs) -> Future[R]:
         self._thread_pool
         return self._thread_pool.submit(fn, *args, **kwargs)
 
+
+    def _get_token_source(self, token: Optional[Union[bool, str]] = None) -> str:
+        """
+        Determine the source of the token being used.
+        
+        Args:
+            token: The token parameter passed to whoami
+            
+        Returns:
+            str: Source of the token ("parameter", "environment", "file", "colab", or "unknown")
+        """
+        if isinstance(token, str):
+            return "parameter"
+        
+        if os.getenv("HF_TOKEN") is not None:
+            return "environment"
+            
+        token_path = Path.home() / ".cache/huggingface/token"
+        if token_path.exists():
+            return "file"
+            
+        try:
+            from google.colab import auth
+            if auth.has_oauth2_token():
+                return "colab"
+        except ImportError:
+            pass
+            
+        return "unknown"
```

Comment:
Instead of checking directly, could you use `_get_token_from_google_colab`/`_get_token_from_environment`/`_get_token_from_file` defined in https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/utils/_auth.py ? This will be much more robust and future-proof. Otherwise, if tomorrow we change the cache path, we would have to update the logic in 2 places. Also, these methods have additional checks like looking at `HF_HOME` instead of assuming file is in the default location.